pkg/ipcache/metadata.go

```diff
@@ -0,0 +1,406 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2018-2019 Authors of Cilium
+
+package ipcache
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/identity/cache"
+	"github.com/cilium/cilium/pkg/labels"
+	cidrlabels "github.com/cilium/cilium/pkg/labels/cidr"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/source"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// idMDMU protects the IdentityMetadata map.
+	//
+	// If this mutex will be held at the same time as the IPCache mutex,
+	// this mutex must be taken first and then take the IPCache mutex.
+	idMDMU lock.RWMutex
+	// IdentityMetadata maps IP prefixes (x.x.x.x/32) to their labels.
+	IdentityMetadata = make(map[string]labels.Labels)
```

Comment:
this map should be made private at the risk of accessing it without holding the mutex.